### DISCUSSION TITLE
Example request: repo that mixes remark-lint and other ESLint plugins

### DISCUSSION THREAD

**Discussion by JoshuaKGoldberg** [remarkjs/remark#1071] [Q&A] [answered]
I'm experimenting with [remark-lint](https://github.com/remarkjs/remark-lint) here: https://github.com/typescript-eslint/typescript-eslint/pull/6052.

After moving all file settings to ESLint `overrides`, `.md` files are failing with this line:

```plaintext
/home/runner/work/typescript-eslint/typescript-eslint/CODE_OF_CONDUCT.md
  0:0  error  Parsing error: Cannot use import statement outside a module
```

https://github.com/typescript-eslint/typescript-eslint/actions/runs/3510036024/jobs/5879552006

Are there any examples of projects that use both `remark-lint` and `typescript-eslint` together?

**Comment by ChristianMurphy** [comment, marked as answer]
> Example request: repo that mixes remark-lint and other ESLint plugins #1071 

There isn't an official example.
But you could check out https://github.com/ChristianMurphy/json-schema-keyref which is a repo I maintain which uses both.

---

I would note, the error you have shared:

> ```
> /home/runner/work/typescript-eslint/typescript-eslint/CODE_OF_CONDUCT.md
>  0:0  error  Parsing error: Cannot use import statement outside a module
> ```

Appears to be entirely contained within `eslint`/`typescript-eslint`/`eslint-plugin-mdx`.
`remark-lint` happens to be in the dependency list, but doesn't appear to be directly or indirectly involved with that lint check.

---

It also appears that `eslint-plugin-mdx` to check pure Markdown files, without any MDX?
You are welcome to do so, though if you only need markdown, and don't currently use MDX features (JSX in Markdown) or plan to in the near future.
It may be worth considering using https://github.com/eslint/eslint-plugin-markdown which is also builds on `remark` for markdown parsing.

If you do plan on using MDX, see the configuration guide at https://github.com/mdx-js/eslint-mdx and ensure you have the configuration setup as needed.
If you believe the issue is related to that plugin, you're welcome to provide a reproduction and open an issue at https://github.com/mdx-js/eslint-mdx

---

Last but not least, if you do plan on using MDX.
It appears you are using Docusaurus, which is still on MDX 1 (at the time of this discussion, see: https://github.com/facebook/docusaurus/issues/4029 for the current progress towards an MDX 2 upgrade)
While linting files with MDX version 2, which could also cause some incompatibility.
You may want to give the docusaurus upgrade to MDX 2 issue a nudge forward.
Or downgrade the `eslint-plugin-mdx` version you are using so the MDX versions are the same.

**Reply by JoshuaKGoldberg** [reply]
Thanks! This is great info 🙂.

FWIW I filed:
* https://github.com/eslint/eslint/issues/16565 around improving the experience around debugging parsing issues.
* https://github.com/mdx-js/eslint-mdx/issues/441 for improving this specific error message

Edit: https://github.com/typescript-eslint/typescript-eslint/pull/6052 has a generally working config too.
